### ISSUE TITLE
2019-7-30

### ISSUE THREAD

**Issue by ssfc** [ssfc/wangdao#71]
221.Read data from MNIST data; can we store them in a numpy array? Investigate the appearance of X and y; done; copy relative data from python to tensorflow; deal with load data; done; success; delete extra comments; done; (2019-6-23)
222.To evaluate it, we had to run init=tf.global_variables_initializer(); To summarize, remember to initialize your variables, create a session and run the operations inside the session;  A placeholder is an object whose value you can specify only later. To specify values for a placeholder, you can pass in values by using a "feed dictionary" (feed_dict variable); the second method does not need to close session; (2019-6-23)
223.In my handwriting, change train set to test set in the predict function; change X and Y from train set; change X and Y to X_train and Y_train; done; change train set to test set in the predict function; change it be more accurate if we raise the iteration? Yes, it is more accurate now; (2019-6-24)
224.Write progress report to Prof He; done; read through tensorflow of Andrew Ng; this article shows us the function one by one; this article also tells us we can place tensorflow in a function; num_px * num_px = 64 * 64 * 3 = 12288; tf.contrib.layers.xavier_initializer; W1 and b1 are both combined in theta1; is the cost really equals to our own? (2019-6-24)
225.Separate W and b is our hand-writing program; strange; we must modify X_train; change X_train to X_test; it becomes right; let us change back; change iteration to 60; okay, done; let us first work on test set, it is better; (2019-6-24)
226.ops.reset_default_graph(); give up; tf.set_random_seed; Sets the graph-level random seed for the default graph; it seems that write a new is easier than modify an old; the program reports that No module named 'tf_utils'; check the functions in tf_utils; load_dataset; we can write our own; random_mini_batches; hold; convert_to_one_hot; we can write our own; predict; we can write our own; what is h5py; hold; (2019-6-24)
227.Let us try previous function of tensorflow; can we change profile in coursera? Give up; we can see that coursera is transpose of our matrix; we should first load the data; done; if predict accuracy is higher than 10%, then it works; do we need to transpose X and Y? As we can see, they print after finishing loading; yes, transpose; done; (2019-6-24)
228.Deal with forward prop; done; deal with cost; done; deal with model; deal with minibatches; done; we should place accuracy in the iteration; change learning rate to see whether we can get better result; obviously wrong, the rate should be higher than 10; let us try sigmoid; yes, it become larger than 10; change hidden layer size; give up; (2019-6-24)
229.Buy an remote control; review neural network in machine learning; it is important to know the shape of all function; find the how to do research pdf; read guide of MIT; get really a lot; buy socks; done; review gradient explode and gradient vanish; read MIT graduate entrance guide;
